scm/etc/scripts/gmtb_scm_analysis.py

```diff
@@ -259,6 +268,26 @@ def print_progress(n_complete, n_total):
     h_advec_thil.append(nc_fid.variables['h_advec_thil'][:])
     h_advec_qt.append(nc_fid.variables['h_advec_qt'][:])
     T_s.append(nc_fid.variables['T_s'][:])
+    try:
+        T_soil_1.append(nc_fid.variables['T_soil_1'][:])
+        T_soil_2.append(nc_fid.variables['T_soil_2'][:])
+        T_soil_3.append(nc_fid.variables['T_soil_3'][:])
+        T_soil_4.append(nc_fid.variables['T_soil_4'][:])
+        soil_moisture_1.append(nc_fid.variables['soil_moisture_1'][:])
+        soil_moisture_2.append(nc_fid.variables['soil_moisture_2'][:])
+        soil_moisture_3.append(nc_fid.variables['soil_moisture_3'][:])
+        soil_moisture_4.append(nc_fid.variables['soil_moisture_4'][:])
+    except KeyError:
```

Comment:
So this means that if soil temperature is there (but not soil moisture), both variables get set to missing values, correct? I assume this is what you want. Also, this is hardcoded to four levels. Any plan to generalize this? RUC runs are writing our 9 levels.

Reply:
This is a quick and dirty check. In practice, they are always written out together in the SCM, so this check is sufficient. There are plans to drop these plotting scripts entirely if/when I ever get around to implementing something more user-friendly as part of the HSUP project. So, no, I don't have plans for generalizing soil levels. If someone uses this with RUC/9 levels, they will have to modify the code.